File: smart-hub/src/main/kotlin/io/github/tassiLuca/hub/coroutines/Launcher.kt
package io.github.tassiLuca.hub.coroutines

import io.github.tassiLuca.hub.coroutines.infrastructure.MockedHubManager
import kotlinx.coroutines.runBlocking

/** The application entry point. */
fun main(): Unit = runBlocking {
    println("Main thread is ${Thread.currentThread().name}")
    MockedHubManager(this.coroutineContext).run()
}

File: smart-hub/src/main/scala/io/github/tassiLuca/hub/Launcher.scala
package io.github.tassiLuca.hub

import gears.async.default.given
import gears.async.Async
import io.github.tassiLuca.hub.infrastructure.MockedHubManager

/** The launcher of a mocked version of the application, using UI simulators. */
@main def launchUIMockedHub(): Unit =
  Async.blocking:
    MockedHubManager().run()
